src/sentry/tasks/unmerge.py

```diff
@@ -273,6 +280,27 @@ def truncate_denormalizations(group):
     features.delete(group)
 
 
+def collect_group_environment_data(events):
+    results = {}
+    for event in events:
+        results[(event.group_id, get_environment_name(event))] = event.get_tag('sentry:release')
+    return results
+
+
+def repair_group_environment_data(caches, project, events):
```

Comment:
I didn't test this explicitly (the behavior is pretty simple), but it's covered by the general unmerge test.